source/Account.cpp

```diff
@@ -329,116 +324,300 @@ void Account::SetSalaryIncome(const string &name, int64_t amount)
 
 
 
-int64_t Account::CrewSalariesOwed() const
+string Account::Step(int64_t assets, int64_t salaries, int64_t maintenance)
 {
-	return crewSalariesOwed;
+	ostringstream out;
+
+	// Step 1: Pay the bills
+	std::vector<Receipt> receipts = PayBills(salaries, maintenance);
+	out << GenerateMissedPaymentLogs(&receipts);
+
+	// If any mortgage has been fully paid off, remove it from the list.
+	UpdateMortgages();
+
+	AddHistory(CalculateNetWorth(assets));
+	UpdateCreditScore(&receipts);
+
+	// If you didn't make any payments, no need to continue further.
+	if(!AnyPaymentsMade(&receipts))
+		return out.str();
+
+	out << GeneratePaymentLogs(&receipts);
+	return out.str();
 }
 
 
 
-void Account::PaySalaries(int64_t amount)
+std::vector<Receipt> Account::PayBills(int64_t salaries, int64_t maintenance)
```

Comment:
This function name seems to be a little too generic, and a little bit off (at least for my native country/language):
- Salary payments are accompanied by a payslip. Only contractors typically send a bill to get their salary. Others just get a payslip from their employer without sending any bill by themselves.
- Fines are just sent as fine (we don't call them bill, we just call them fine)
- Mortgage payments are typically automatically deducted from bank-accounts. Banks don't send a bill, but they do send receipts (and when needed a reminder).

I would recommend to move the content from this function into Step(...) It is only 4 functions and all those four functions have very clear, specific, well-defined names.

(I believe that moving the content of this function into Step() and then dropping this function also addresses some of the scrolling concerns which quyykk has.)

Reply:
Resolved with 4ce7945